handler/src/test/java/io/netty/handler/ssl/SslHandlerTest.java

```diff
@@ -1214,11 +1245,18 @@ public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) {
                                 }
                             });
                         }
-                    }).connect(sc.localAddress());
+                    }).connect(serverAddress);
             cc = future.syncUninterruptibly().channel();
 
-            assertTrue(clientSslHandler.handshakeFuture().await().isSuccess());
-            assertTrue(serverSslHandler.handshakeFuture().await().isSuccess());
+            assertTrue(clientSslHandler.handshakeFuture().sync().isSuccess());
+
+            ReferenceCountedOpenSslEngine engine = (ReferenceCountedOpenSslEngine) clientSslHandler.engine();
+            // This test only works for non TLSv1.3 as TLSv1.3 will establish sessions after
+            // the handshake is done.
+            // See https://www.openssl.org/docs/man1.1.1/man3/SSL_CTX_sess_set_get_cb.html
+            if (!SslUtils.PROTOCOL_TLS_V1_3.equals(engine.getSession().getProtocol())) {
```

Comment:
IIRC the session reuse for TLSv1.3 should work in the same was as session id from the OpenSSL API point of view, only time of receiving PSK is different. Are you going to work on session resumption for TLSv1.3 in a follow-up?

Reply:
it already works... I just need to come up with a unit test as you can not assume the session is sent when the handshake completes.